include/circt/Dialect/SSP/SSPPasses.td

```diff
@@ -20,4 +20,15 @@ def PrintInstance : Pass<"ssp-print-instance", "mlir::ModuleOp"> {
   let constructor = "circt::ssp::createPrintInstancePass()";
 }
 
+def Schedule : Pass<"ssp-schedule", "mlir::ModuleOp"> {
```

Comment:
Could this run on ssp instance ops rather than mlir::ModuleOp? It should be able to, and then it could run on them in parallel.

Reply:
In principle, yes, the instances are all independent. Unfortunately, I can't do it at the moment, because I construct a completely new `ssp::InstanceOp`, so I then would have to replace the pass' root operation. I'll change the pass once I implement the smarter approach (i.e. just updating the property attributes).

Reply:
Ah I thought for some reason that a pass on an ssp::InstanceOp should be allowed to replace the root ssp::InstanceOp, but it appears that is illegal according to the docs on this:

> This includes adding or removing operations from the parent block